contrib/agent/src/main/java/io/opencensus/contrib/agent/ContextStrategyImpl.java

```diff
@@ -22,8 +24,27 @@
  */
 final class ContextStrategyImpl implements ContextStrategy {
 
+  private final Cache<Thread, Context> savedContexts
+          = CacheBuilder.newBuilder().weakKeys().build();
```

Comment:
Added a comment to #savedContexts. Re reuse of thread IDs: no - thread IDs don't matter in this implementation. If using thread IDs as keys (not doing that), it should still be safe as thread IDs generated by incrementing a long in Thread.